File: src/main/kotlin/GitHubContent.kt
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class GitHubContent(
    @SerialName("_links")
    val links: Links,
    @SerialName("download_url")
    val downloadUrl: String?,
    @SerialName("git_url")
    val gitUrl: String,
    @SerialName("html_url")
    val htmlUrl: String,
    val name: String,
    val path: String,
    val sha: String,
    val size: Long,
    val type: String,
    val url: String
)

@Serializable
data class Links(
    val git: String,
    val html: String,
    val self: String
)

File: dart/lib/github_content/github_content.dart
// ignore_for_file: invalid_annotation_target

import 'package:freezed_annotation/freezed_annotation.dart';

part 'github_content.freezed.dart';
part 'github_content.g.dart';

@freezed
class GitHubContent with _$GitHubContent {
  const factory GitHubContent({
    @JsonKey(name: '_links')
     required Links links,
    @JsonKey(name: 'download_url')
    required String? downloadUrl,
    @JsonKey(name: 'git_url')
    required String gitUrl,
    @JsonKey(name: 'html_url')
    required String htmlUrl,
    required String name,
    required String path,
    required String sha,
    required int size,
    required String type,
    required String url,
  }) = _GitHubContent;

  factory GitHubContent.fromJson(Map<String, Object?> json) => _$GitHubContentFromJson(json);
}

@freezed
class Links with _$Links {
  const factory Links({
    required String git,
    required String html,
    required String self,
  }) = _Links;

  factory Links.fromJson(Map<String, Object?> json) => _$LinksFromJson(json);
}